### PR TITLE
Setup the sphinx and generated the sdk API docs

### LINKED ISSUE #2373: [GSOC] Poject 7 Tracking Issue: Automating Documentation for katib Python SDK
### Objective
Automate the Katib Python SDK reference documentation generation and render it on `kubeflow.org`.

### Tasks
- [x] **Proposal Submission:** Submit the GSOC project proposal as a project proposal in Katib.
- [x] **Review Existing Documentation:** Analyze documentation from other projects for best practices.
- [x] **Local Katib Setup:** Clone Katib repository for kickoff project development.
- [x] **Evaluate Documentation Tools:** Compare Sphinx, MkDocs, Jupyter Book, etc., to choose the best tool for Katib's needs.
- [x] **Configure and Test Documentation Tool:** Set up Sphinx (or chosen tool) for Katib. Could you make sure of the quality and readability of the generated documentation? ([#2392](https://github.com/kubeflow/katib/pull/2392))
- [ ] **Integrate with Kubeflow's Website Repo:** Make the Katib repository a submodule of the website repository and test documentation sharing.
- [ ] **Automate Documentation Updates:** 
  - Write a GitHub Action to automate documentation updates and trigger documentation sharing to the website repository.
  - Script to handle events and update website files.

### Expected Outcomes
- Fully automated, high-quality documentation process.
- Up-to-date documentation on the Katib website.

### PR DESCRIPTION
**Pull request by shivas1516**
**What this PR does / why we need it**:
This PR sets the Sphinx configuration for generating the Katib SDK API reference documentation. It also includes updates to `gen_python.py` to build the documentation in Markdown format.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes  #2373

**Checklist:**

- [] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

**Comment by shivas1516** [comment]
@andreyvelich @shivaylamba 
In this PR, I have configured Sphinx and modified some of the docstrings in the `sdk/python/v1beta1/kubeflow/katib/api/katib_client.py` file to ensure consistency and compatibility with Sphinx for generating documentation. Please let me know if any further changes are needed.

**Review by shivaylamba — COMMENTED** [review]
Can you resolve the merge conflicts in the PR @shivas1516 

Also do you plan to create a separate PR for integration with Github actions pipelines for automated documentation updates?

File: hack/gen-python-sdk/post_gen.py [hunk 1]
```diff
@@ -164,6 +165,42 @@ def update_python_sdk(src, dest, versions=('v1beta1')):
     # Clear working dictionary
     shutil.rmtree(src)
 
+def generate_sphinx_docs(src, dest):
+    
+    project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+
+    docs_dir = os.path.join(project_root, 'docs', 'sdkdocs')
```
Comment (shivaylamba) [inline hack/gen-python-sdk/post_gen.py:172]: Currently we are usinhardcoded paths, does it makes sense to use environment variables instead for better configuration? @andreyvelich let me know your thoughts

Comment (nsingl00) [reply hack/gen-python-sdk/post_gen.py:172]: +1. Having environment variable would be Better for easy configuration management.

**Comment by nsingl00** [comment]
@shivas1516 Is there a way to preview the content generated by sphinx? Can you add the preview screenshot in description if you can see it locally?

**Comment by andreyvelich** [comment]
@shivas1516 Do you want to to open a new PR with your change ?

**Comment by shivas1516** [comment]
> @shivas1516 Do you want to to open a new PR with your change ?

I thought that, what do you think, @andreyvelich? Your thoughts would be appreciated.

Thanks!

**Comment by andreyvelich** [comment]
> > @shivas1516 Do you want to to open a new PR with your change ?
> 
> I thought that, what do you think, @andreyvelich? Your thoughts would be appreciated.
> 
> Thanks!

It's ok, but you can just push the changes in this PR.

**Pull request closed without merging**